internal/dataplane/parser/translate_httproute_test.go

```diff
@@ -1479,3 +1371,22 @@ func TestIngressRulesFromHTTPRoutes_RegexPrefix(t *testing.T) {
 func HTTPMethodPointer(method string) *gatewayv1beta1.HTTPMethod {
 	return (*gatewayv1beta1.HTTPMethod)(&method)
 }
+
+func k8sObjectInfoOfHTTPRoute(route *gatewayv1beta1.HTTPRoute) util.K8sObjectInfo {
+	// parsers always provide the annotations map
+	anotations := make(map[string]string)
```

Comment:
nit: we annotate here regardless. If `oute.Annotations != nil` then we don't need to

Reply:
I've reversed the condition and assigments.